CMakeLists.txt

```diff
@@ -64,86 +68,104 @@ endif()
 
 ### config.h shenanigans
 
-file(READ src/config.h.in CONFIG_CONTENTS)
-string(REPLACE "#mesondefine" "#cmakedefine" CONFIG_CONTENTS "${CONFIG_CONTENTS}")
-string(REPLACE "#cmakedefine C_TARGETCPU" "#cmakedefine C_TARGETCPU @C_TARGETCPU@" CONFIG_CONTENTS "${CONFIG_CONTENTS}")
-string(REPLACE "#cmakedefine CUSTOM_DATADIR" "#cmakedefine CUSTOM_DATADIR \"@CUSTOM_DATADIR@\"" CONFIG_CONTENTS "${CONFIG_CONTENTS}")
+set(TEST_CODE_BUILTIN_AVAILABLE "
+int main() {
+    if (__builtin_available(macOS 11, *)) {
+        return 0;
+    }
+    return 0;
+}
+")
```

Comment:
Interesting we can do this!

Reply:
I read that a code snippet compile test is preferable to doing a symbol check when it's a built-in/intrinsic/etc.